cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsPlatformResources.java

```diff
@@ -901,6 +905,30 @@ public CloudEncryptionKeys encryptionKeys(ExtendedCloudCredential cloudCredentia
         return cloudEncryptionKeys;
     }
 
+    private Set<KeyListEntry> getListKeyResult(AmazonKmsClient client) {
+        String nextMarker = null;
+        Set<KeyListEntry> kmsKeys = new HashSet<>();
+        do {
+            ListKeysRequest listKeysRequest = new ListKeysRequest().withMarker(nextMarker);
+            ListKeysResult listKeysResult = client.listKeys(listKeysRequest);
+            kmsKeys.addAll(listKeysResult.getKeys());
```

Comment:
What does `listKeysResult.getKeys()` return? Is it a `List<>` or `Set<>`? I wonder if `KeyListEntry` implements `hashCode()` and `equals()` properly, otherwise the usage of a `Set<>` for `kmsKeys` does not really make sense.

Reply:
The listKeysResult.getKeys() returns a list of `List<KeyListEntry> `

Reply:
Okay, so what is the conclusion then? Is it reasonable to collect the results in a set? What about the `hashCode()` and `equals()` of `KeyListEntry`?

Reply:
Apologies for the delay here. Since the AWS call for listKeysResult.getKeys() returns a List like I mentioned above i.e. 
List<KeyListEntry> , It makes more sense to collect the kmskeys as a list as well. I have made the changes accordingly and have tested this as well. Thanks for pointing out this.